src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -983,7 +983,7 @@ private int ReadResponseContent(HttpResponseMessage response, Span<byte> buffer)
                 while (buffer.Length != 0)
                 {
                     // Sync over async here -- QUIC implementation does it per-I/O already; this is at least more coarse-grained.
-                    if (_responseDataPayloadRemaining <= 0 && !ReadNextDataFrameAsync(response, CancellationToken.None).GetAwaiter().GetResult())
+                    if (_responseDataPayloadRemaining <= 0 && !ReadNextDataFrameAsync(response, CancellationToken.None).AsTask().GetAwaiter().GetResult())
```

Comment:
What makes the original incorrect?

Reply:
`ReadNextDataFrameAsync` returns a `ValueTask`. Thou shalt never try to access the result state of a `ValueTask`/`ValueTask<T>` until it's known to have completed.